include/GoogleSync/GoogleSync.php

```diff
@@ -678,11 +788,36 @@ public function pullEvent($event_remote, $event_local = null)
             $private = array();
         }
 
-        $private['suitecrm_id'] = $event->id;
-        $private['suitecrm_type'] = $event->module_name;
+        $private['suitecrm_id'] = $event_local->id;
+        $private['suitecrm_type'] = $event_local->module_name;
 
         $extendedProperties->setPrivate($private);
 
+        return $extendedProperties;
+    }
+
+    /**
+     * Pull event from Google Calendar to SuiteCRM
+     *
+     * If the SuiteCRM Meeting is not provided, a new one will be created
+     * and inserted. If one is provided, the existing meeting will be updated.
+     *
+     * @param \Google_Service_Calendar_Event $event_remote \Google_Service_Calendar_Event Object
+     * @param Meeting $event_local Meeting (optional) \Meeting Bean
+     *
+     * @return bool Success/Failure of setLastSync, since that's what saves the record
+     */
+    public function pullEvent(Google_Service_Calendar_Event $event_remote, Meeting $event_local = null)
+    {
+        if (!isset($event_local) || empty($event_local)) {
+            $event = $this->createSuitecrmMeetingEvent($event_remote);
+        } else {
+            $event = $this->updateSuitecrmMeetingEvent($event_local, $event_remote);
```

Comment:
$event could be boolean false if error occured in updateSuitecrmMeetingEvent(). Please handle the error.

Reply:
Yep... that means something horrible happened. Now logging it and returning false from the function, which gets handled downstream.